language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -682,24 +927,41 @@ impl DependencyGraph {
         name: PM::PackageName,
         dependency_cache: &mut DependencyCache,
         external_requests: &mut Vec<ExternalRequest>,
+        overrides: &mut BTreeMap<PM::PackageName, Package>,
         progress_output: &mut Progress,
-    ) -> Result<()> {
+    ) -> Result<Package> {
         let pkg = match self.package_table.entry(name) {
             Entry::Vacant(entry) => entry.insert(pkg),
 
-            // Seeing the same package again, pointing to the same dependency: OK, return early.
+            // Seeing the same package again, pointing to the same dependency: OK, return early but
+            // only if seeing a package that was not on an "override path" (was created when no
+            // override was active); otherwise we need to keep inspecting the graph to make sure
+            // that the overrides introduced on this path correctly dominate all "uses" of a given
+            // package
             Entry::Occupied(entry) if entry.get() == &pkg => {
-                return Ok(());
+                if entry.get().overridden_path {
+                    // check if acyclic to avoid infinite recursion
+                    self.check_acyclic()?;
+                    self.override_verify(&pkg, name, overrides)?;
+                }
+                return Ok(pkg);
```

Comment:
Should this be guarded by the `else` of the `if` above?  The comment above seems to suggest so.

This also implies a missing test case.

Reply:
> Should this be guarded by the `else` of the `if` above? The comment above seems to suggest so.

This was not my intention. The idea was that if  package already in the graph (the same as package we are trying to insert)  is NOT on the upgrade path, we are OK to return early. Otherwise, either `override_verify` will return an error or not, in which case it is also OK to return (early as there are no more packages to analyze and insert here).

Reply:
Got it, that makes sense, thanks for clarifying!